src/audio/volume/volume_hifi5.c

```diff
@@ -0,0 +1,516 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2024 Intel Corporation. All rights reserved.
+//
+// Author: Andrula Song <andrula.song@intel.com>
+
+/**
+ * \file
+ * \brief Volume HiFi5 processing implementation without peak volume detection
+ * \authors Andrula Song <andrula.song@intel.com>
+ */
+
+#include <sof/audio/buffer.h>
+#include <sof/audio/component.h>
+#include <sof/common.h>
+#include <ipc/stream.h>
+#include <stddef.h>
+#include <stdint.h>
+
+LOG_MODULE_DECLARE(volume, CONFIG_SOF_LOG_LEVEL);
+
+#include "volume.h"
+
+#if SOF_USE_HIFI(5, VOLUME)
+
+#if (!CONFIG_COMP_PEAK_VOL)
+
+#include <xtensa/tie/xt_hifi5.h>
+
+/**
+ * \brief store volume gain 4 times for xtensa multi-way intrinsic operations.
+ * Simultaneous processing 4 samples.
+ * \param[in,out] cd Volume component private data.
+ * \param[in] channels_count Number of channels to process.
+ */
+static void vol_store_gain(struct vol_data *cd, const int channels_count)
+{
+	int32_t i;
+
+	/* using for loop instead of memcpy_s(), because for loop costs less cycles */
```

Comment:
this loop can be replaced with a single `memcpy()` and you've found out that the loop is faster?.. Interesting, then we have a problem with our `memcpy()`